File: Java/Problem_003.java
/**
 * 
 * @author Christian Seely
 *
 */
public class Problem_003 {

	public static long problemThree()
	{
		long max = 0;
		int div = 2;
		long n = 600_851_475_143L;
		while(true)
		{
			if((div*div) > n)
			{
				if(n != 1 && n > max) max = n;
				break;
			}
			if(n % div == 0)
			{
				if(div > max) max = div;
				n /= div;
				while(n % div == 0) n /= div;
			}
			div++;
		}
		return max;
	}

	public static void main(String[] args) { System.out.println(problemThree()); }
}

File: Haskell/Problem_003.hs
-- Author: Christian Seely

primeFactors :: Integer -> [Integer]
primeFactors = factorize 2
  where 
    factorize a n
        | n == 1 = []
        | a * a > n = [n]
        | rem == 0 = a : factorize a div
        | otherwise = factorize (a + 1) n
        where
            (div, rem) = divMod n a

problemThree :: Integer
problemThree = last $ primeFactors 600851475143

main = print $ problemThree